cs3/app/provider/v1beta1/provider_api.proto

```diff
@@ -53,11 +53,38 @@ import "cs3/types/v1beta1/types.proto";
 // Any method MAY return UNKNOWN.
 // Any method MAY return UNAUTHENTICATED.
 service ProviderAPI {
-  // Returns the App provider URL
+  // Creates a new file to be opened by an app. The new file MAY be either
+  // empty or generated out of a template if the AppProvider has such capability.
+  rpc CreateFileForApp(CreateFileForAppRequest) returns (CreateFileForAppResponse);
+  // Returns the App URL and all necessary info to open a resource in an online editor.
   // MUST return CODE_NOT_FOUND if the resource does not exist.
   rpc OpenInApp(OpenInAppRequest) returns (OpenInAppResponse);
 }
 
+message CreateFileForAppRequest {
+  // OPTIONAL.
+  // Opaque information.
+  cs3.types.v1beta1.Opaque opaque = 1;
+  // REQUIRED.
+  // The target container where the file has to be created.
+  cs3.storage.provider.v1beta1.Reference ref = 2;
+  // REQUIRED.
+  // The name of the file to be created.
+  string filename = 3;
+}
```

Comment:
You can add an optional field named "template". Looking forward in case we want to provide custom templates (like CERN branded Collabora files).

Reply:
Makes sense, as eventually the user chooses between e.g. `New Word Document` or `New Word Document from corporate template`...